tests/func/test_data_cloud.py

```diff
@@ -894,3 +894,79 @@ def test(self):
         assert self.message_header in self._caplog.text
         assert self.message_foo_part in self._caplog.text
         assert self.message_bar_part in self._caplog.text
+
+
+# New pytest style tests parametrized properly
+
+import pytest  # noqa
+from dvc import api  # noqa
+
+
+# NOTE: staticmethod is only needed in Python 2
+class S3:
+    should_test = staticmethod(_should_test_aws)
+    get_url = staticmethod(get_aws_url)
+
+
+class GCP:
+    should_test = staticmethod(_should_test_gcp)
+    get_url = staticmethod(get_gcp_url)
+
+
+class Azure:
+    should_test = staticmethod(_should_test_azure)
+    get_url = staticmethod(get_azure_url)
+
+
+class OSS:
+    should_test = staticmethod(_should_test_oss)
+    get_url = staticmethod(get_oss_url)
+
+
+class SSH:
+    should_test = staticmethod(_should_test_ssh)
+    get_url = staticmethod(get_ssh_url)
+
+
+class HDFS:
+    should_test = staticmethod(_should_test_hdfs)
+    get_url = staticmethod(get_hdfs_url)
+
+
+remote_params = [S3, GCP, Azure, OSS, SSH, HDFS]
+
+
+@pytest.fixture
+def remote(request):
+    if not request.param.should_test():
+        raise pytest.skip()
+    return request.param
+
+
+def pytest_generate_tests(metafunc):
+    if "remote" in metafunc.fixturenames:
+        metafunc.parametrize("remote", remote_params, indirect=True)
+
+
+def test_api_get_url(repo_dir, dvc_repo, remote):
+    remote_url = remote.get_url()
+
+    main(["remote", "add", "-d", "upstream", remote_url])
+    dvc_repo.add(repo_dir.FOO)
+
+    assert api.get_url(repo_dir.FOO) == "%s/%s" % (
+        remote_url,
+        "ac/bd18db4cc2f85cedef654fccc4a4d8",
+    )
+
+
+def test_api_open(repo_dir, dvc_repo, remote):
```

Comment:
If you are testing "api", maybe just create a new test file instead of appending all of this to test_data_cloud? Might even consider adding a directory like `tests/func/remote/`.

Reply:
As long as I use all those `_should_test()` and `_get_url()` it's far easier to have it here. Once `DataCloud` is gone this could be separated into `test_remotes.py` and `test_api.py`.

Reply:
I am just being incremental here.

Reply:
@Suor Nothing prevents you from using _should_test and _get_url in a test_api.py. No need to make test_data_cloud.py even worse than it is already :slightly_smiling_face: